android/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt

```diff
@@ -342,18 +496,65 @@ fun getPromotionalOffer() : ErrorContainer {
 
 // region private functions
 
+internal fun mapStringToProductType(type: String) : ProductType {
+    MappedProductCategory.values()
+        .firstOrNull { it.value.equals(type, ignoreCase = true) }
+        ?.let {
+            return it.toProductType
+        }
+
+    // Maps strings used in deprecated hybrid methods to native ProductType enum
+    // "subs" and "inapp" are legacy purchase types used in v4 and below
+    return when(type.lowercase()) {
+        "subs" -> ProductType.SUBS
+        "inapp" -> ProductType.INAPP
+        else -> {
+            warnLog("Unrecognized product type: $type... Defaulting to INAPP")
+            ProductType.INAPP
+        }
+    }
+}
+
+internal class InvalidProrationModeException(): Exception()
+
+@Throws(InvalidProrationModeException::class)
+internal fun getGoogleProrationMode(prorationMode: Int?) : GoogleProrationMode?  {
```

Comment:
Some of these could be private but are made internal for tests right? In those cases we should add `@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)`. That way we could get some linter errors if trying to use these somewhere else.